libraries/Wire/src/utility/twi.c

```diff
@@ -271,11 +286,16 @@ uint8_t twi_writeTo(uint8_t address, uint8_t* data, uint8_t length, uint8_t wait
     TWCR = _BV(TWINT) | _BV(TWEA) | _BV(TWEN) | _BV(TWIE);	// enable INTs, but not START
   }
   else
-    // send start condition
+  // send start condition
     TWCR = _BV(TWINT) | _BV(TWEA) | _BV(TWEN) | _BV(TWIE) | _BV(TWSTA);	// enable INTs
 
   // wait for write operation to complete
+  twi_timeout(1);  
   while(wait && (TWI_MTX == twi_state)){
+    if (twi_timeout(0))
+	{
```

Comment:
Please follow the formatting style of the rest of the code in the file and attach the brace to the `if` statement.